tests/providers/amazon/aws/operators/test_redshift_to_s3.py

```diff
@@ -76,3 +76,49 @@ def test_execute(self, table_as_file_name, expected_s3_key, mock_run, mock_sessi
 
         assert mock_run.call_count == 1
         assert_equal_ignore_multiple_spaces(self, mock_run.call_args[0][0], unload_query)
+
+    @parameterized.expand([
+        [True, "key/None_"],
+        [False, "key"],
```

Comment:
I like that you used parameterized. But we need to add custom_select_query to parameterized parameters and also we need to handle the failure cases here (with various combinations of None parameters set - just table None, just schema None, all None etc so the parameterized should have the "expected exception" also (or maybe we should have failure test cases as separate method - both will work.

Reply:
@potiuk Got it! Will work on it now